user-console/src/main/java/org/pentaho/mantle/client/ui/tabs/MantleTabPanel.java

```diff
@@ -94,6 +175,45 @@ public void addTab( String text, String tooltip, boolean closeable, Widget conte
     if ( getSelectedTab() == null ) {
       selectTab( tab );
     }
+    MantleTabMenuItem menuItem = createTabMenuItem( tab );
+    menuItemHashMap.put( tab, menuItem );
+    tabsSubMenuBar.addItem( menuItem );
+    contextMenuRefreshThreshold( true );
+    updateTabMenuText( tab );
+  }
+
+  public void renameMenuTab( PentahoTab tab ){
+    menuItemHashMap.get( tab ).setText( tab.getLabelText() );
```

Comment:
Regardless of the comment questioning the hash map, it would be nice to have a `getMenuItem(PentahoTab)` method that would abstract how the mapping is achieved.